runtime.c

```diff
@@ -153,38 +153,36 @@ bool check_nt(bam1_t *read, int32_t position, unsigned char nt, bool exact)
 	if (read->core.flag & BAM_FUNMAP) {
 		return false;
 	}
-	if (read->core.pos > position || compute_mapped_end(read) < position) {
+	if (read->core.pos +1 > position || compute_mapped_end(read) < position) {
 		return false;
 	}
-	if ((read->core.flag & BAM_FUNMAP) || read->core.n_cigar == 0) {
-		mapped_position =
-		    (read->core.flag & BAM_FREVERSE) ? (read->core.l_qseq -
-							position +
-							read->core.pos -
-							1) : (position -
-							      read->core.pos);
+
+	if ( read->core.n_cigar == 0) {
+			mapped_position = bam_is_rev(read) ? (read->core.l_qseq - position + read->core.pos) : (position - read->core.pos-1);
 	} else {
-		int32_t required_offset =
-		    (read->core.flag & BAM_FREVERSE) ? (read->core.l_qseq -
-							position +
-							read->core.pos -
-							1) : (position -
-							      read->core.pos);
+		int32_t required_offset = bam_is_rev(read) ? (read->core.l_qseq - position + read->core.pos) : (position - read->core.pos-1);
 		mapped_position = 0;
-
-		for (int index = 0;
-		     index < read->core.n_cigar && required_offset > 0;
-		     index++) {
-			uint8_t consume =
-			    bam_cigar_type(bam_cigar_op
-					   (bam_get_cigar(read)[index]));
-			if (consume & 2)
-				required_offset--;
-			if (consume & 1)
-				mapped_position++;
+		/* Consume cigar operations one at a time */
+		for(int index = 0;index < read->core.n_cigar; index++) {
+			uint8_t consume = bam_cigar_type(bam_cigar_op(bam_get_cigar(read)[index]));
+			uint32_t op_length = bam_cigar_oplen(bam_get_cigar(read)[index]);
+			
+			/* Iterate through the length of the cigar operation */
+			for(int incr = 0; incr < op_length && required_offset >= 0; incr++){
```

Comment:
Please don't call this `incr`. Perhaps `index` should get renamed to `cigar_index` and then `incr` to `op_index`.